src/rebuild/srv.c

```diff
@@ -1148,6 +1227,61 @@ rebuild_leader_start(struct ds_pool *pool, uint32_t rebuild_ver,
 	return rc;
 }
 
+static int
+rebuild_task_complete_schedule(struct rebuild_task *task, struct ds_pool *pool,
+			       struct rebuild_global_pool_tracker *rgt)
+{
+	int rc;
+
+	if (rgt == NULL) {
+		rc = ds_rebuild_schedule(pool, task->dst_map_ver, ++task->dst_rebuild_gen,
+					 &task->dst_tgts, task->dst_rebuild_op, task->dst_max_ver,
+					 5);
+		return rc;
+	}
+
+	/* If current job failed */
+	if (!is_rebuild_global_done(rgt) || rgt->rgt_status.rs_errno != 0) {
+		rgt->rgt_status.rs_state = DRS_IN_PROGRESS;
+		if (task->dst_rebuild_op == RB_OP_RECLAIM) {
+			rc = ds_rebuild_schedule(pool, task->dst_map_ver, ++task->dst_rebuild_gen,
+						 &task->dst_tgts, RB_OP_RECLAIM, task->dst_max_ver,
+						 5);
+			return rc;
+		}
+
+		/* Schedule reclaim to clean up current op */
+		rc = ds_rebuild_schedule(pool, task->dst_map_ver, ++task->dst_rebuild_gen,
+					 &task->dst_tgts, RB_OP_RECLAIM, task->dst_max_ver, 5);
+		if (rc)
+			return rc;
+
+		/* Then retry */
+		rc = ds_rebuild_schedule(pool, task->dst_map_ver, ++task->dst_rebuild_gen,
+					 &task->dst_tgts, task->dst_rebuild_op, task->dst_max_ver,
+					 5);
+		return rc;
+	}
+
+	/* Schedule reclaim for reintegrate/extend/upgrade to cleanup stale object */
+	if (task->dst_rebuild_op == RB_OP_REINT || task->dst_rebuild_op == RB_OP_EXTEND) {
+		rgt->rgt_status.rs_state = DRS_IN_PROGRESS;
+		rc = ds_rebuild_schedule(pool, task->dst_map_ver, ++task->dst_rebuild_gen,
+					 &task->dst_tgts, RB_OP_RECLAIM, task->dst_max_ver, 5);
+		if (rc != 0)
+			D_ERROR("reschedule reclaim, "DF_UUID" failed: "DF_RC"\n",
+				DP_UUID(task->dst_pool_uuid), DP_RC(rc));
```

Comment:
need not retry/reschedule the rebuild task here?

Reply:
this is succeed case, so it does not need retry, only schedule the reclaim to cleanup the stale objects.

Reply:
ok, thx